### PR TITLE
fix: Do not compute Brier score when predict_proba is not available

### LINKED ISSUE #1050: Bug for `_get_scorers_to_add` with binary classification with classifier that expose `predict_proba`
The `get_scorers_to_add` for binary classification will compute the `brier_score` without checking that the `estimator` has a `predict_proba` method.

So testing a binary classification problem with `SVC` should fail when requesting the default metric.

I reworking this part in `EstimatorReport` but it means that we are missing a non-regression test for this case currently in the `CrossValidationReporter`.

### PR DESCRIPTION
**Pull request by glemaitre**
closes #1050 

As specified, do not try to compute Brier score with an estimator that does not provide probability estimate (i.e. does not have a `predict_proba` method).

Added a non-regression test as well.

### CODE REVIEW

File: skore/tests/unit/sklearn/test_cross_validate.py [hunk 1]
```diff
@@ -35,3 +40,66 @@ def test_cross_validate_return_estimator():
...
+        ),
+        # Multiclass classification without predict_proba
+        (
+            SVC(probability=False),
+            make_classification,
+            {"n_classes": 3, "n_clusters_per_class": 1},
+            {"recall_weighted", "precision_weighted"},
+        ),
+    ],
+    ids=[
```
Comment (augustebaum) [inline skore/tests/unit/sklearn/test_cross_validate.py:89]: Use [`pytest.param`](https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids) to keep the IDs closer to the examples, or create `case_` functions instead (see https://github.com/probabl-ai/skore/blob/29d3689920a77176f92d3dd18ceb52397882b247/skore/tests/unit/sklearn/train_test_split/test_train_test_split.py#L20)

**Review by augustebaum — CHANGES_REQUESTED** [review]

**Comment by augustebaum** [comment]
Nice work, just a suggestion for the `parametrize`

**Review by augustebaum — APPROVED** [review]
Nice!
